src/Aspire.Hosting/build/Aspire.Hosting.targets

```diff
@@ -104,14 +98,83 @@ internal static class ]]>%(ClassName)<![CDATA[
     <Warning Code="ASPIRE001" Text="The '$(Language)' language isn't fully supported by Aspire - some code generation targets will not run, so will require manual authoring." />
   </Target>
 
+  <!--
+  Gets the references in 'AppProjectTargetFramework' that aren't executable projects.
+  -->
+  <UsingTask TaskName="GetNonExecutableReferences"
+             TaskFactory="RoslynCodeTaskFactory"
+             AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll">
+    <ParameterGroup>
+      <AppProjectTargetFramework ParameterType="Microsoft.Build.Framework.ITaskItem[]" Required="true" />
+      <NonExecutableReferences ParameterType="Microsoft.Build.Framework.ITaskItem[]" Output="true" />
+    </ParameterGroup>
+    <Task>
+      <Using Namespace="System.Xml.Linq" />
+      <Code Type="Fragment" Language="cs">
+        <![CDATA[
```

Comment:
@rainersigwald  what's the perf cost of inline tasks these days? Originally they used the old CodeDOM and were super slow to compile and not cached.

Reply:
On my machine, when it was “cold” with no node reuse, the task took 700ms. On a warm incremental build it took 70ms.

I plan on logging an issue to move this task into a .dll, but didn’t want to block on that infrastructure.

Reply:
Cool. Maybe they do it in memory now, or cache.

Reply:
Logged https://github.com/dotnet/aspire/issues/1825